osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -492,6 +496,12 @@ public void Dispose()
             onlineLookupQueue?.Dispose();
         }
 
+        TextureStore IBeatmapResourceProvider.LargeTextureStore => largeTextureStore;
+        ITrackStore IBeatmapResourceProvider.Tracks => trackStore;
+        AudioManager IStorageResourceProvider.AudioManager => audioManager;
+        IResourceStore<byte[]> IStorageResourceProvider.Files => Files.Store;
+        IResourceStore<TextureUpload> IStorageResourceProvider.CreateTextureLoaderStore(IResourceStore<byte[]> underlyingStore) => host.CreateTextureLoaderStore(underlyingStore);
```

Comment:
Above suggestion shows that this has an unchecked access to possibly-null host. Only caller (`LegacySkin`) is null-tolerant, so I'd null-check here and call it a day.

```suggestion
        IResourceStore<TextureUpload> IStorageResourceProvider.CreateTextureLoaderStore(IResourceStore<byte[]> underlyingStore) => host?.CreateTextureLoaderStore(underlyingStore);
```